### ISSUE TITLE
False positive for methods containing anonymous classes

### ISSUE THREAD

**Issue by marcospassos** [pepakriz/phpstan-exception-rules#34]
The following example reports an incorrect and conflicting warning:

```php
class Example extends TestCase
{
    /**
     * @throws \OutOfBoundsException
     */
    public function testName() : void
    {
        $foo = new class {
            public function foo() : string {
                return 'foo';
            }
        };

        $foo = new Foo();

        $foo->throw();
    }
}

class Foo
{
    /**
     * @throws \OutOfBoundsException
     */
    public function throw() : void
    {
        throw new \OutOfBoundsException();
    }
}
```
Warning:
```
 ------ ------------------------------------------------- 
  Line   Example.php                                      
 ------ ------------------------------------------------- 
  15     Unused @throws OutOfBoundsException annotation   
  25     Missing @throws OutOfBoundsException annotation  
 ------ ------------------------------------------------- 
```
I'd to be happy to provide a PR with the fix if you can give some guidance on how to fix it  :)

**Comment by pepakriz** [comment]
I know about this limitation, but I was lazy to implement it :)

Try current master. Your use-case should works. If you'll find any other broken case, please create next issue.

Thank you!

**Comment by marcospassos** [comment]
Thank you for fixing it :)
